document-store/src/integrationTest/java/org/hypertrace/core/documentstore/utils/CreateUpdateTestThread.java

```diff
@@ -0,0 +1,98 @@
+package org.hypertrace.core.documentstore.utils;
+
+import java.io.IOException;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.core.documentstore.CreateResult;
+import org.hypertrace.core.documentstore.Document;
+import org.hypertrace.core.documentstore.Filter;
+import org.hypertrace.core.documentstore.Filter.Op;
+import org.hypertrace.core.documentstore.SingleValueKey;
+import org.hypertrace.core.documentstore.UpdateResult;
+
+public class CreateUpdateTestThread extends Thread {
```

Comment:
should this be final? Also, could we elaborate why this class is needed?

Reply:
will make it final. Used in concurrency tests - e.g `testUpdateWithMultipleThreads`